core/cas-server-core-tickets/src/test/java/org/apereo/cas/ticket/registry/DefaultTicketRegistryTests.java

```diff
@@ -77,8 +85,28 @@ void verifyCountForService() throws Throwable {
     void verifyEncodeFails() throws Throwable {
         val cipher = new DefaultTicketCipherExecutor(null, null,
             "AES", 512, 16, "webflow");
-        val reg = new DefaultTicketRegistry(cipher, mock(TicketSerializationManager.class), new DefaultTicketCatalog());
+        val reg = new DefaultTicketRegistry(cipher, mock(TicketSerializationManager.class), new DefaultTicketCatalog(),
+                mock(ConfigurableApplicationContext.class));
         assertNull(reg.encodeTicket(null));
         assertNotNull(reg.decodeTicket(mock(Ticket.class)));
     }
+
+    @RepeatedTest(1)
+    void verifyGetExpiredTicket() throws Throwable {
+        val applicationContext = mock(ConfigurableApplicationContext.class);
+        val registry = new DefaultTicketRegistry(mock(TicketSerializationManager.class), new DefaultTicketCatalog(),
+                applicationContext);
+        val originalAuthn = CoreAuthenticationTestUtils.getAuthentication();
+        val ticketGrantingTicket = new TicketGrantingTicketImpl(TestTicketIdentifiers.generate().ticketGrantingTicketId(),
+                originalAuthn, new HardTimeoutExpirationPolicy(1));
+        registry.addTicket(ticketGrantingTicket);
+        Thread.sleep(1500);
```

Comment:
This is suspect; Can you consider using awaitility? There is a good chance this would fail randomly in CI.

Reply:
This is suspect yes ;-) but I don't think we have a problem here. There is no async processing on which we wait. We create a TGT expiring in one second. We wait 1,5 seconds. When we get it, it is expired. Always.